### PR TITLE
Migrate FCI tests to pytest

### PR DESCRIPTION
**Pull request by gerritholl**
Migrate FCI tests to pytest.  This enables the more flexible set of
fixtures available there.  This will be needed by at least one of the
tests to be added in https://github.com/pytroll/satpy/pull/1177

 - [ ] Closes #xxxx 
 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``flake8 satpy`` 
 - [ ] Fully documented

### CODE REVIEW

**Comment by gerritholl** [comment]
CodeFactor is wrong to complain here.  [PEP 8](https://www.python.org/dev/peps/pep-0008/) prohibits a *bare* test of `if len(x)`.  It does not prohibit an explicit length test such as `if len(x) > 0`:

> ```
> # Correct:
> if not seq:
> if seq:
> ```

> ```
> # Wrong:
> if len(seq):
> if not len(seq):
> ```

Since https://github.com/PyCQA/pylint/pull/2815 (apparently not merged in the version CodeFactor uses), pylint reportedly no longer complains about explicit comparisons of `len(x) > 0`.  Therefore, I propose this particular recommendation should be ignored.

File: satpy/tests/reader_tests/test_fci_l1c_fdhsi.py [hunk 1]
```diff
@@ -426,13 +419,11 @@ def test_handling_bad_data_ir(self):
...
         reader.create_filehandlers(loadables)
-        with self.assertLogs(
-                'satpy.readers.fci_l1c_fdhsi',
-                level="ERROR") as cm:
+        with caplog.at_level("ERROR"):
             reader.load([DatasetID(
                     name="ir_123",
                     calibration="brightness_temperature")])
-        self.assertRegex(cm.output[0], "cannot produce brightness temperatur")
+            assert "cannot produce brightness temperatur" in caplog.text
```
Comment (mraspaud) [inline satpy/tests/reader_tests/test_fci_l1c_fdhsi.py:429]:
```suggestion
            assert "cannot produce brightness temperature" in caplog.text
```

Comment (gerritholl) [reply satpy/tests/reader_tests/test_fci_l1c_fdhsi.py:429]: That typo was already there before :)

Comment (mraspaud) [reply satpy/tests/reader_tests/test_fci_l1c_fdhsi.py:429]: I think @ameraner is fixing it in his PR

**Comment by gerritholl** [comment]
The test is failing because the previous test contained a bug, which I serendipitously fixed:

https://github.com/pytroll/satpy/blob/a82e4a61453dc98336106f347a76ed3fadd3d030/satpy/tests/reader_tests/test_fci_l1c_fdhsi.py#L261

This should have been `assertEqual`, and it would have failed, because we have `len(files) == 5`.  In the new version it is

https://github.com/gerritholl/satpy/blob/7a8a121bce2a81e6af68652239f293edafde96cd/satpy/tests/reader_tests/test_fci_l1c_fdhsi.py#L256

which is correct.

File: satpy/tests/reader_tests/test_fci_l1c_fdhsi.py (hunk 1 above)
Comment (ameraner) [reply satpy/tests/reader_tests/test_fci_l1c_fdhsi.py]: Yes I do! See https://github.com/pytroll/satpy/pull/1135/files#diff-78bf1063f7ffb6655291b1b2aa2c06a0R466

Comment (gerritholl) [reply satpy/tests/reader_tests/test_fci_l1c_fdhsi.py]: We're likely to see merge conflicts between those PRs.

**Pull request merged**